sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroGenericCoder.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.coders;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+
+/** GenericRecord Avro Coder for simple dict type cross-language data transfer. */
+public class AvroGenericCoder extends AvroCoder<GenericRecord> {
+  protected AvroGenericCoder(Schema schema) {
```

Comment:
Also, should we be adding this to standard coder definitions in [1] and [2].

[1] https://github.com/apache/beam/blob/master/model/fn-execution/src/main/resources/org/apache/beam/model/fnexecution/v1/standard_coders.yaml
[2] https://github.com/apache/beam/blob/master/model/pipeline/src/main/proto/beam_runner_api.proto#L557

Reply:
I think we can follow this up in a separate issue.
https://issues.apache.org/jira/browse/BEAM-7309